File: src/main/java/org/antop/java/chapter4/PolyRoll.java
package org.antop.java.chapter4;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

public class PolyRoll {
    public static final PolyRoll INSTANCE = new PolyRoll();
    private final List<Employee> allEmployees = new ArrayList<>();

    private PolyRoll() {}

    public void add(Employee employee) {
        allEmployees.add(employee);
    }

    public void calculateSalary() {
        BigDecimal totalSalary = BigDecimal.ZERO;
        for (Employee employee : allEmployees) {
            totalSalary = totalSalary.add(employee.salary());
        }
        System.out.println("totalSalary = " + totalSalary);
    }

    public static void main(String[] args) {
        PolyRoll.INSTANCE.add(new Employee("2p", BigDecimal.valueOf(2000)));
        PolyRoll.INSTANCE.add(new Employee("Antop", BigDecimal.valueOf(1000)));
        PolyRoll.INSTANCE.add(new Employee("MJ", BigDecimal.valueOf(500)));
        PolyRoll.INSTANCE.calculateSalary();
    }
}

File: src/main/kotlin/org/antop/kotlin/chapter4/PolyRoll.kt
package org.antop.kotlin.chapter4

import java.math.BigDecimal

object PolyRoll {
    private val allEmployees = mutableListOf<Employee>()

    fun add(employee:Employee) {
        allEmployees += employee
    }

    fun calculateSalary() {
        var totalSalary = BigDecimal.ZERO
        for (p in allEmployees) {
            totalSalary += p.salary
        }
        println("totalSalary = $totalSalary")
    }

}

fun main() {
    PolyRoll.add(Employee("2P", BigDecimal.valueOf(2000)))
    PolyRoll.add(Employee("Antop", BigDecimal.valueOf(1000)))
    PolyRoll.add(Employee("MJ", BigDecimal.valueOf(500)))
    PolyRoll.calculateSalary()
}
